### PR TITLE
[Frontend] Support for chat completions input in the tokenize endpoint

### PR DESCRIPTION
**Pull request by sasha0552**
Related: #5054 (https://github.com/vllm-project/vllm/pull/5054#issuecomment-2195039364)

Example:

```
$ curl -H "Authorization: Bearer token-abc123" -H "Content-Type: application/json" -d '{"model":"meta-llama/Meta-Llama-3-8B-Instruct","prompt":"test"}' http://localhost:8000/tokenize
{
  "count": 1,
  "max_model_len": 8192,
  "tokens": [1985],
}

$ curl -H "Authorization: Bearer token-abc123" -H "Content-Type: application/json" -d '{"model":"meta-llama/Meta-Llama-3-8B-Instruct","messages":[{"role":"user","content":"Hi"}]}' http://localhost:8000/tokenize
{
  "count": 11,
  "max_model_len": 8192,
  "tokens": [128000,128006,882,128007,271,13347,128009,128006,78191,128007,271]
}

$ curl -H "Authorization: Bearer token-abc123" -H "Content-Type: application/json" -d '{"model":"meta-llama/Meta-Llama-3-8B-Instruct","tokens":[128000,128006,882,128007,271,13347,128009,128006,78191,128007,271]}' http://localhost:8000/detokenize
{
  "prompt": "<|begin_of_text|><|start_header_id|>user<|end_header_id|>\n\nHi<|eot_id|><|start_header_id|>assistant<|end_header_id|>\n\n"
}
```

### CODE REVIEW

**Comment by DarkLight1337** [comment]
Hmm, I think we can move this to a new file since it's not really a part of OpenAI's Chat or Completions API.

File: tests/entrypoints/openai/test_tokenization.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "model_name",
+    [MODEL_NAME],
+)
+async def test_tokenize_chat(client: openai.AsyncOpenAI, model_name: str):
+    base_url = str(client.base_url)[:-3].strip("/")
+    tokenizer = get_tokenizer(tokenizer_name=MODEL_NAME, tokenizer_mode="fast")
```
Comment (sasha0552) [inline tests/entrypoints/openai/test_tokenization.py:47]:
```suggestion
    tokenizer = get_tokenizer(tokenizer_name=model_name, tokenizer_mode="fast")
```

**Review by sasha0552 — COMMENTED** [review]
Just a reminder for myself (so I don't cancel the CI).

**Review by DarkLight1337 — APPROVED** [review]
LGTM, thanks for extending this feature!

**Pull request merged**